docs/architecture/api-packages.md

```diff
@@ -0,0 +1,164 @@
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+In a result, the `api-packages` label value is validated. Also, the API Package name uniqueness within Application is guaranteed. Event and API Definition existence is not checked, apart from verifying the UUID type of the ID arrays.
+
+This schema cannot be modified by user. We validate this JSON schema against internal JSON schema, similarly to `scenarios` LabelDefinition.
+
+### Assumptions
+
+- API or Event Definition can be a part of API Package only within the same Application entity.
+- API Package name doesn't have to be unique within tenant.
+
+### Pros
+
+- No breaking changes in our API
+- Consistent implementation (identical to Scenarios)
+- Static label validation
+
+### Cons
+
+- Additional logic on client side (see the [Client Usage](#client-usage) section)
+- _(not important)_ No validation for existence of API/Event Definition IDs
+
+## Example
+
+> **NOTE:** For clarity, in this example IDs has been changed from UUID type to string.
+
+1. User creates Application `sample-app`.
+1. User registers three different API definitions with the following IDs: `foo`, `bar` and `baz`.
+1. In order to create API Package called `sample-package`, User sets the following label for `sample-app` Application:
+
+   **Key**: `api-packages`
+   **Value:**
+
+   ```json
+   {
+     "sample-package": {
+       "description": "A Sample API Package",
+       "metadata": {
+         "displayName": "Sample Package",
+         "documentationURL": "",
+         "supportURL": ""
+       },
+       "events": [],
+       "apis": ["bar", "baz"]
+     }
+   }
+   ```
+
+1. Runtime Agent configures Runtime.
+1. In Service Catalog, there are two Service Classes: `foo` and `sample-package`, which contains single Service Plan with `bar` and `baz` APIs grouped.
+
+## Client Usage
+
+The following section describes client usage of the API Packages.
+
+### Runtime Agent
+
+To create Service Classes in Kyma Service Catalog from Application APIs and Event Definitions, Runtime Agent (or another component on Runtime side) has to populate the Service Class list.
+
+In order to do so, Runtime Agent executes the following steps:
+
+1. Query Applications for Runtime
+1. For every Application:
+   - Read `api-packages` label and populate ServiceClasses for API Packages. Agent generates IDs for these Service Classes.
+   - Read all API and Event Definitions, excluding items from all API Packages. Every API or Event Definition is a Separate Class with ID equal to API/Event Definition ID.
+
+### UI
+
+To display all API Packages, UI executes the following steps:
+
+1. Get all Applications
+1. Read `api-packages` label value from every Application (if it does exist)
+1. Display API Packages
+
+As API Package name doesn't have to be unique within tenant, UI displays Application name for every API Package, to distinguish API Packages with equal names.
+
+## Future
```

Comment:
Please mention, that in the future we can support validation between api-packages and apis, eventAPIs, in bboth directions.